config/routes.rb

```diff
@@ -7,6 +7,10 @@
 
   post '/companies/update_companies_list', to: 'companies#update_companies_list'
   post '/companies/update_companies_details', to: 'companies#update_companies_details'
+  post '/companies/:id/update_shareholders', to: 'companies#update_company_shareholders',
+    as: 'companies_update_company_shareholders'
+  post '/companies/update_shareholders', to: 'companies#update_selected_companies_shareholders',
+    as: 'companies_update_selected_companies_shareholders'
```

Comment:
I would consider merging it into a single endpoint

Reply:
Looks good, I merge it.